core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala

```diff
@@ -160,8 +175,6 @@ private[spark] object CachedEventHubsReceiver extends CachedReceiver with Loggin
   def apply(ehConf: EventHubsConf,
             nAndP: NameAndPartition,
             startSeqNo: SequenceNumber): CachedEventHubsReceiver = {
-    val cr = new CachedEventHubsReceiver(ehConf, nAndP)
-    cr.createReceiver(startSeqNo)
-    cr
+    new CachedEventHubsReceiver(ehConf, nAndP, startSeqNo)
```

Comment:
Nice one

Reply:
Yea, since the seqNos (and receive calls) aren't being made one at a time any more, there's actually more cleaning I can do. I'll try to get rid of the `null` receiver and clean this a bit more tomorrow.